src/secret_storage.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the SecretStorage for managing the persistent secret storage for the Flask charm."""
+
+import secrets
+import typing
+
+from ops import Object, Relation, RelationDataContent, RelationEvent
+
+if typing.TYPE_CHECKING:
+    from charm import FlaskCharm
+
+
+class SecretStorage(Object):
+    """A class that manages secret keys required by the FlaskCharm.
+
+    Attrs:
+        is_initialized: True if the SecretStorage has been initialized.
+    """
+
+    # bandit think this is a password
+    _FLASK_SECRET_KEY_KEY = "flask_secret_key"  # nosec
```

Comment:
Could these be module level constants?

Reply:
Since the module only have one class, I think putting in the class is okay?